### PR TITLE
portable: clarify error when detach fails to match to existing unit

### PR DESCRIPTION
**Pull request by bluca**
Clarify that it's looking at attached units, not at units inside the referenced
image. Also take into account extensions.

### CODE REVIEW

File: src/portable/portable.c [hunk 1]
```diff
@@ -1674,8 +1675,20 @@ int portable_detach(
...
+        log_debug("No unit files associated with '%s%s%s' found attached to the system. Image not attached?",
+                  name_or_path,
+                  isempty(extensions) ? "" : "' or any of its extensions '",
+                  isempty(extensions) ? "" : extensions);
+        return sd_bus_error_setf(error,
+                                 BUS_ERROR_NO_SUCH_UNIT,
+                                 "No unit files associated with '%s%s%s' found attached to the system. Image not attached?",
+                                 name_or_path,
+                                 isempty(extensions) ? "" : "' or any of its extensions '",
+                                 isempty(extensions) ? "" : extensions);
```
Comment (poettering) [inline src/portable/portable.c]: please reuse the message. i.e.:

```c
r = sd_bus_error_setf(error, …);
log_debug_errno(r, "%s", error->message);
return r;
```

i.e. no need to have the very same format string in place twice here.

Comment (bluca) [reply src/portable/portable.c]: pre-existing condition :-) but sure, fixed

**Comment by keszybz** [comment]
LGTM.

File: src/portable/portable.c [hunk 2]
```diff
@@ -1674,8 +1675,18 @@ int portable_detach(
...
+                return -ENOMEM;
+
+        r = sd_bus_error_setf(error,
+                              BUS_ERROR_NO_SUCH_UNIT,
+                              "No unit files associated with '%s%s%s' found attached to the system. Image not attached?",
+                              name_or_path,
+                              isempty(extensions) ? "" : "' or any of its extensions '",
+                              isempty(extensions) ? "" : extensions);
+        log_debug_errno(r, "%s", error->message);
+        return r;
```
Comment (poettering) [inline src/portable/portable.c]: actually you can merge these two lines even:

```c
return log_debug_errno(r, "%s", error->message);
```

Comment (bluca) [reply src/portable/portable.c]: done

**Comment by poettering** [comment]
lgtm

**Pull request merged**